scrapbook/encoders.py

```diff
@@ -0,0 +1,170 @@
+        ):
+            raise ScrapbookDataException(
+                "Data load request for {encoder} is not a JSON type."
+                "Instead got '{data_type}'".format(
+                    encoder=scrap.encoder, data_type=type(scrap.data)
+                )
+            )
+        loader = self._encoders.get(scrap.encoder)
+        if not loader:
+            raise ScrapbookException(
+                'No encoder found for "{}" encoder type!'.format(scrap.encoder)
+            )
+        return loader.decode(scrap, **kwargs)
+
+    def encode(self, scrap, **kwargs):
+        """
+        Finds the register for the given encoder and translates the scrap's data
+        from an object of the encoder type to a JSON typed object.
+
+        Parameters
+        ----------
+        scrap: Scrap
+            A partially filled in scrap with data that needs encoding
+        """
+        encoder = self._encoders.get(scrap.encoder)
+        if not encoder:
+            raise ScrapbookException(
+                'No encoder found for "{data_type}" data type!'.format(
+                    data_type=encoder
+                )
+            )
+        output_scrap = encoder.encode(scrap, **kwargs)
+        if not isinstance(
+            scrap.data,
+            tuple([list, dict] + list(six.integer_types) + list(six.string_types)),
+        ):
+            raise ScrapbookDataException(
+                "Data encoded from '{encoder_type}' is not a JSON type."
+                "Instead got '{data_type}'".format(
+                    encoder_type=type(encoder), data_type=type(scrap.data)
+                )
+            )
+        return output_scrap
+
+
+class JsonEncoder(object):
+    def encode(self, scrap, **kwargs):
+        if isinstance(scrap.data, six.string_types):
+            scrap = scrap._replace(data=json.loads(scrap.data))
+        return scrap
+
+    def decode(self, scrap, **kwargs):
+        # Just in case we somehow got a valid JSON string pushed
+        if isinstance(scrap.data, six.string_types):
+            scrap = scrap._replace(data=json.loads(scrap.data))
+        return scrap
+
+
+class TextEncoder(object):
+    def encode(self, scrap, **kwargs):
+        if not isinstance(scrap.data, six.string_types):
+            # TODO: set encoder information to save as encoding
+            scrap = scrap._replace(data=str(scrap.data))
+        return scrap
+
+    def decode(self, scrap, **kwargs):
+        # Just in case we somehow got a non-string saved?!
+        if not isinstance(scrap.data, six.string_types):
+            # TODO: read from encoder information to load as encoding
+            scrap = scrap._replace(data=str(scrap.data))
+        return scrap
+
+
+class ArrowDataframeEncoder(object):
+    def encode(self, scrap, **kwargs):
+        pass  # Implement eventually
+
+    def decode(self, scrap, **kwargs):
+        pass  # Implement eventually
```

Comment:
Since this class is here but not used, I'd either not include this _or_ at the very least have it `raise Exception("Not implemented")`